clients/goredis.go

```diff
@@ -1,16 +1,20 @@
 package clients
 
 import (
+	"context"
 	"fmt"
-	goredis "github.com/go-redis/redis/v7"
-	"github.com/nitishm/go-rejson/rjs"
 	"strings"
+
+	goredis "github.com/go-redis/redis/v8"
+	"github.com/nitishm/go-rejson/rjs"
 )
 
+var ctx = context.Background()
```

Comment:
I really don't think using a global `context` is good here, as it will be of no use for anyone using `go-redis@v8`. I think a better approach will be when we can propagate a new context with the commands themselves. 

But changing the interface to use context will drop the backward compatibility. And we also support `redigo` client, which does not use `context` right now.

Reply:
What would be an alternate strategy to achieving this? The `v8` api is not compatible with `v7` and below without a mechanism to provision a `context`.

Reply:
https://github.com/nitishm/go-rejson/pull/46#discussion_r546385774